text/src/autogluon/text/automm/optimization/lit_module.py

```diff
@@ -174,7 +178,12 @@ def training_step(self, batch, batch_idx):
         -------
         Average loss of the mini-batch data.
         """
-        output, loss = self._shared_step(batch)
+        if self.mixup_fn is not None and (self.hparams.mixup_off_epoch is None or self.current_epoch < self.hparams.mixup_off_epoch):
+            batch, target_a, target_b, lam = self.mixup_fn(batch, "timm_image_image", self.model.label_key)
+            output, _ = self._shared_step(batch)
+            loss = self._compute_loss(output=output, label=target_a) * lam + self._compute_loss(output=output,label=target_b) * (1. - lam)
+        else:
+            output, loss = self._shared_step(batch)
```

Comment:
This design has limitations since it uses a fixed model prefix `timm_image_image`. Users may customize this key and there are maybe multiple image backbones, e.g., `timm_image_0`, `timm_image_1`. Multimodal models, e.g., CLIP, also has its image backbone. 

To make mixup scalable to these situations, I think we should move mixup into the model definitions, e.g, timm_image.py and clip.py. We can let the model's output also include the label information, e.g., {"label": {"label_a": label_a, "label_b": label_b}}. Then, in `lit_module.py`, we index label from each model's output to compute the loss and metric, instead of using the label from `batch`. To make this design consistent across all models, we also need to modify the return of other models to include its label information.